### PR TITLE
[Releases][MedApplication] Adding Med Application support for windows.

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
This allows the MedApplication to be compiled in windows. Also added the necessary files to be able to release it as a pip package.

Ping @Marco1410 because you use salome
Also ping @philbucher @sunethwarna you can find how to compile the med in win in the dockerfile.

**🆕 Changelog**
Med compilable in win

### CODE REVIEW

**Review by sunethwarna — APPROVED** [review]
Thanks a lot @roigcarlo. I hope @philbucher  is also ok with this :)

File: applications/MedApplication/custom_io/med_model_part_io.cpp [hunk 1]
```diff
@@ -362,6 +362,9 @@ class MedModelPartIO::MedFileHandler
         // read only by default, unless other settings are specified
         med_access_mode open_mode;
 
+        // Fix to allow windows conversion from whatever eldritch format is using to something convertible to c_str()
+        std::string med_file_mame{rFileName.string()};
```
Comment (philbucher) [inline applications/MedApplication/custom_io/med_model_part_io.cpp:366]: mame? 😄

**Review by philbucher — APPROVED** [review]
Great job 👏🏻

File: applications/MedApplication/custom_io/med_model_part_io.cpp [hunk 2]
```diff
@@ -371,7 +374,9 @@ class MedModelPartIO::MedFileHandler
             // basic checks if the file is compatible with the MED library
             med_bool hdf_ok;
             med_bool med_ok;
-            const med_err err = MEDfileCompatibility(rFileName.c_str(), &hdf_ok, &med_ok);
+
+            const med_err err = MEDfileCompatibility(med_file_mame.c_str(), &hdf_ok, &med_ok);
```
Comment (philbucher) [inline applications/MedApplication/custom_io/med_model_part_io.cpp:378]: maybe `rFileName.string().c_str()` would also work :man_shrugging:

Comment (roigcarlo) [reply applications/MedApplication/custom_io/med_model_part_io.cpp:378]: not it does not :D

File: applications/MedApplication/custom_io/med_model_part_io.cpp (hunk 1 above)
Comment (roigcarlo) [reply applications/MedApplication/custom_io/med_model_part_io.cpp:366]: a f**k, now I see it now. I`ll Fix it later :P

**Pull request merged**